packages/thumbprint-codemods/src/avatar-sizes/test.js

```diff
@@ -0,0 +1,167 @@
+    });
+
+    test('`size` defined outside the `Avatar` and within another `Avatar`', () => {
+        const codeBefore = `import React from 'react';
+        import { Avatar, Input } from '@thumbtack/thumbprint-react';
+        const size = "large";
+        const App = (props) => (
+            <div>
+                <Avatar size={size} />
+                <Avatar size="medium" />
+                <Input />
+            </div>
+        )`;
+
+        const codeAfter = `import React from 'react';
+        import { Avatar, Input } from '@thumbtack/thumbprint-react';
+        const size = "large";
+        const App = (props) => (
+            <div>
+                <Avatar size={size} />
+                <Avatar size="small" />
+                <Input />
+            </div>
+        )`;
+
+        expect(transform(codeBefore)).toBe(codeAfter);
+        expect(console.error).toHaveBeenCalledTimes(1);
+    });
+
+    test.only('props are spread and also has a large size', () => {
+        const codeBefore = `import React from 'react';
+        import { Avatar, Input } from '@thumbtack/thumbprint-react';
+        const App = (props) => (
+            <div>
+                <Avatar {...props} size="large" />
+                <Avatar {...props} />
+                <Input />
+            </div>
+        )`;
+
+        const codeAfter = `import React from 'react';
+        import { Avatar, Input } from '@thumbtack/thumbprint-react';
+        const App = (props) => (
+            <div>
+                <Avatar {...props} size="medium" />
+                <Avatar {...props} />
+                <Input />
+            </div>
+        )`;
+
+        expect(transform(codeBefore)).toBe(codeAfter);
+        expect(console.error).toHaveBeenCalledTimes(1);
+    });
+
+    test('one "large" size and one "xlarge" size', () => {
+        const codeBefore = `import React from 'react';
+        import { Avatar } from '@thumbtack/thumbprint-react';
+        const App = (props) => (
+            <>
+                <Avatar size="xlarge" />
+                <Avatar size="large" />
+            </>
+        )`;
+
+        const codeAfter = `import React from 'react';
+        import { Avatar } from '@thumbtack/thumbprint-react';
+        const App = (props) => (
+            <>
+                <Avatar size="xlarge" />
+                <Avatar size="medium" />
+            </>
+        )`;
+
+        expect(transform(codeBefore)).toBe(codeAfter);
+        expect(console.error).toHaveBeenCalledTimes(1);
+    });
+});
+
+describe('output should not change', () => {
+    test('from a different package', () => {
+        const code = `
+        import React from 'react';
+        import { Avatar } from './my-avatar.jsx';
+        const App = (props) => (
+            <>
+                <Avatar size="large" />
+                <Avatar size="medium" />
+                <Avatar size="small" />
+            </>
+        )`;
+
+        expect(transform(code)).toBeNull();
```

Comment:
Should this be `null`?

We do it will `null` in the `Button` test but not in the other `Avatar` test:
https://github.com/thumbtack/thumbprint/blob/master/packages/thumbprint-codemods/src/avatar-import-name/test.js#L75-L95

I'm not sure which one is correct.

Reply:
yeah, the behaviour is different. the other avatar codemod returns the source code even when it doesn't change anything, whereas this one and the button one return null. should probably update the avatar name codemod to have the same behaviour as here, but that's a different PR

Reply:
Cool, just wanted to see which was one correct.